models/Message.php

```diff
@@ -289,23 +288,16 @@ protected static function makeCacheKey()
     }
 
     /**
-     * Creates a sterile key for a message.
+     * Creates an md5 representation of the message Id
      * @param  string $messageId
      * @return string
      */
-    protected static function makeMessageCode($messageId)
+    public static function makeMessageCode($messageId)
     {
-        $separator = '.';
-
-        // Convert all dashes/underscores into separator
-        $messageId = preg_replace('!['.preg_quote('_').'|'.preg_quote('-').']+!u', $separator, $messageId);
-
-        // Remove all characters that are not the separator, letters, numbers, or whitespace.
-        $messageId = preg_replace('![^'.preg_quote($separator).'\pL\pN\s]+!u', '', mb_strtolower($messageId));
-
-        // Replace all separator characters and whitespace by a single separator
-        $messageId = preg_replace('!['.preg_quote($separator).'\s]+!u', $separator, $messageId);
-
-        return Str::limit(trim($messageId, $separator), 250);
+        if (Config::get('rainlab.translate::caseSensitiveIds', false)) {
+            return md5(trim($messageId));
+        } else {
+            return md5(mb_strtolower(trim($messageId)));
+        }
```

Comment:
Why was this change required? Also wasn't the previous behavior to be case sensitive? Wouldn't you want these codes to be case sensitive?

Reply:
@LukeTowers previous behavior was case INSENSITIVE and is preserved by default. See old code below:
```
        $messageId = preg_replace('![^'.preg_quote($separator).'\pL\pN\s]+!u', '', mb_strtolower($messageId));
```

Reply:
The change is not REQUIRED but gives the option to developpers to change the behavior.

Reply:
Why would we want to provide the option when we haven't up to this point? Personally I'd think that case sensitive should have been the default from the start, but now that it isn't, what's the benefit of allowing it to be?

Reply:
You kind of just answered that. If you think this should have been the default, why not give that option now? It doesn't hurt anything.

Reply:
I'd be concerned about the message ID changing whenever the config item value changes